Detections/MultipleDataSources/NewUserAgentLast24h.yaml

```diff
@@ -30,35 +35,39 @@ query: |
 
   let starttime = 14d;
   let endtime = 1d;
-  let UserAgentAll =
+  let UserAgentAll = 
   (union isfuzzy=true
   (OfficeActivity
   | where TimeGenerated >= ago(starttime)
-  | where isnotempty(ExtendedProperties)
-  | where ExtendedProperties has "useragent"
-  | extend UserAgent = extractjson("$[0].Value", ExtendedProperties, typeof(string))
   | where isnotempty(UserAgent)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated) by UserAgent, SourceIP = ClientIP, Account = UserId, Type, RecordType, Operation
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by UserAgent, SourceIP = ClientIP, Account = UserId, Type, RecordType, Operation
   ),
   (
   W3CIISLog
   | where TimeGenerated >= ago(starttime)
   | where isnotempty(csUserAgent)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated) by UserAgent = csUserAgent, SourceIP = cIP, Account = csUserName, Type, sSiteName, csMethod, csUriStem
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by UserAgent = csUserAgent, SourceIP = cIP, Account = csUserName, Type, sSiteName, csMethod, csUriStem
   ),
   (
   AWSCloudTrail
   | where TimeGenerated >= ago(starttime)
   | where isnotempty(UserAgent)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated) by UserAgent, SourceIP = SourceIpAddress, Account = UserIdentityUserName, Type, EventSource, EventName
-  ));
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by UserAgent, SourceIP = SourceIpAddress, Account = UserIdentityUserName, Type, EventSource, EventName
+  ))
+  // remove wordSize blocks of non-numeric hex characters prior to word extraction
+  | extend UserAgentNoHexAlphas = replace("([A-Fa-f]{4,})", "x", UserAgent)
+  // once blocks of hex chars are removed, extract wordSize blocks of a-z
+  | extend Tokens = extract_all("([A-Za-z]{4,})", UserAgentNoHexAlphas)
+  // concatenate extracted words to create a summarized user agent for baseline and comparison
+  | extend NormalizedUserAgent = strcat_array(Tokens, "|")
+  | project-away UserAgentNoHexAlphas, Tokens;
   UserAgentAll
```

Comment:
I know this was already put in this way, but could you swap the join around and use leftanti.  Best practices for perf indicate the smaller data set be on the left side of the join  - https://docs.microsoft.com/en-us/azure/kusto/query/joinoperator?pivots=azuredataexplorer

  UserAgentAll
  | where StartTime >= ago(endtime) 
  | summarize StartTime = min(StartTime), EndTime = max(EndTime), count() by UserAgent, NormalizedUserAgent, SourceIP, Account, Type, RecordType, Operation, EventSource, EventName, sSiteName, csMethod, csUriStem
  | join kind=leftanti
  (
  UserAgentAll
  | where StartTime < ago(endtime)
  | summarize by NormalizedUserAgent, SourceIP, Account, Type, RecordType, Operation, EventSource, EventName, sSiteName, csMethod, csUriStem
  ) 
  on NormalizedUserAgent